python/beeai_framework/backend/chat.py

```diff
@@ -80,12 +67,16 @@ class ChatModelStructureOutput(BaseModel):
 
 
 class ChatModelInput(ChatModelParameters):
-    tools: Annotated[list, BeforeValidator(tool_validator)] | None = None
+    tools: list[InstanceOf[Message]] = []
```

Comment:
FYI these updates to `ChatModelStructureInput` and `ChatModelInput` will conflict with #397 so which ever PR is merged second will need to address them